lib/identity/client.js

```diff
@@ -205,6 +206,62 @@ class Client extends PartialClient {
     const newSecretCreated = this.createSecret(newSecret)
     return newSecretCreated
   }
+  /**
+   * shareSecretWithUsername
+   *
+   * @param {String} secretType
+   * @param {String} secretName
+   * @param {Array} usernameToAdd
+   *
+   *
+   */
+  async shareSecretWithUsername(secretName, secretType, usernamesToAdd = []) {
+    // Generate Group to Validate it exists
+    const groupName = `tozny.secret.${this.config.realmName}.${
+      this.storage.config.clientId
+    }.${secretType.toLowerCase()}`
+    const groupList = await this.storage.listGroups(
+      this.storage.config.clientId,
+      [groupName]
+    )
+    if (groupList.groups.length < 1) {
+      throw new Error('Group Does Not Exist')
+    }
+    const currentGroup = groupList.groups[0]
+    // Create the request for username to client id map
+    const SearchRealmIdentitiesRequest = {
+      identity_usernames: usernamesToAdd,
+    }
+    let response = await this.storage.authenticator.tsv1Fetch(
+      `${
+        this.storage.config.apiUrl
+      }/v1/identity/search/realm/${this.config.realmName.toLowerCase()}/identity`,
+      {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(SearchRealmIdentitiesRequest),
+      }
+    )
+    const clientMapping = await validateResponseAsJSON(response)
+    // Create an array and set capabilities to read
+    let groupMembers = []
+    const capabilities = {
+      read: true,
+    }
+    usernamesToAdd.forEach(username => {
+      let usernameLowerCase = username.toLowerCase()
+      let clientID =
```

Comment:
this has a chance of failing if a username provided does not have an associated client. You should verify that it exists.